app/controllers/applicants_controller.rb

```diff
@@ -0,0 +1,35 @@
+class ApplicantsController < ApplicationController
+  def index
+
+  end
+  
+  def show
+    @applicant = Applicant.find(params[:id])
+    @pets = Pet.all
+    
+    if params[:name].present? 
+      @results = @pets.search(params[:name]) 
+      @search = params[:name]
+      # redirect_to "/applicants/#{@applicant.id}"
+    end
```

Comment:
It looks like your button to Add a Pet to an application redirects back to this action using this path: `/applicants/:applicant_id` and then with query parameters for a pet ID

That would mean you have to create your join record in this controller method, which you probably don't want to do (SRP). Another option for that button could be to route it to a `applicant_pets_controller` and specifically do a `create` action there. In that `create` method, you could do something like `PetApplicant.create` with the applicant_id and pet_id. The path for that button could be `/applicants/:applicant_id/pets/:pet_id` or you could also decide to send the pet ID as a hidden field if you didn't want it in the path